File: frontend/types/meeting.ts
import type { SourceCitation } from "@/types/chat";

export type MeetingStatus = "uploaded" | "processing" | "ready" | "failed";

export type MeetingListItem = {
  id: string;
  title: string;
  status: MeetingStatus;
  createdAt: string;
  participantCount: number;
  decisionCount: number;
  actionItemCount: number;
};

export type TranscriptSegment = {
  id: string;
  speaker: string;
  startTime: string;
  endTime?: string | null;
  text: string;
};

export type Decision = {
  id: string;
  text: string;
  source?: SourceCitation | null;
};

export type ActionItem = {
  id: string;
  task: string;
  owner?: string | null;
  dueDate?: string | null;
  source?: SourceCitation | null;
};

export type Risk = {
  id: string;
  text: string;
  source?: SourceCitation | null;
};

export type MeetingDetail = {
  id: string;
  title: string;
  status: MeetingStatus;
  createdAt: string;
  participants: string[];
  summary: string;
  transcript: TranscriptSegment[];
  decisions: Decision[];
  actionItems: ActionItem[];
  risks: Risk[];
};

export type MeetingUploadResponse = {
  id: string;
  title: string;
  status: MeetingStatus;
};

File: backend/app/schemas/meeting.py
from datetime import datetime

from pydantic import BaseModel

from app.schemas.chat import SourceCitation


class MeetingUploadResponse(BaseModel):
    id: str
    title: str
    status: str


class MeetingListItem(BaseModel):
    id: str
    title: str
    status: str
    createdAt: datetime
    participantCount: int
    decisionCount: int
    actionItemCount: int


class DecisionResponse(BaseModel):
    id: str
    text: str
    source: SourceCitation | None = None


class ActionItemResponse(BaseModel):
    id: str
    task: str
    owner: str | None = None
    dueDate: str | None = None
    source: SourceCitation | None = None


class RiskResponse(BaseModel):
    id: str
    text: str
    source: SourceCitation | None = None


class TranscriptSegmentResponse(BaseModel):
    id: str
    speaker: str
    startTime: str
    endTime: str | None = None
    text: str


class MeetingDetailResponse(BaseModel):
    id: str
    title: str
    status: str
    createdAt: datetime
    participants: list[str]
    summary: str
    transcript: list[TranscriptSegmentResponse]
    decisions: list[DecisionResponse]
    actionItems: list[ActionItemResponse]
    risks: list[RiskResponse]
